src/Open3D/IO/FileFormat/FilePCD.cpp

```diff
@@ -618,9 +618,12 @@ bool WritePCDHeader(FILE *file, const PCDHeader &header) {
 
 float ConvertRGBToFloat(const Eigen::Vector3d &color) {
     std::uint8_t rgba[4] = {0, 0, 0, 0};
-    rgba[2] = (std::uint8_t)std::max(std::min((int)(color(0) * 255.0), 255), 0);
-    rgba[1] = (std::uint8_t)std::max(std::min((int)(color(1) * 255.0), 255), 0);
-    rgba[0] = (std::uint8_t)std::max(std::min((int)(color(2) * 255.0), 255), 0);
+    rgba[2] = (std::uint8_t)std::max(
```

Comment:
i found this to be a bit verbose, can we consider doing one of the followings?

- Add a utility function to convert float to uint8 with proper clipping
- Use `std::round(x)` instead of `(int)(x + 0.5)`
- Use some Eigen built-int functions: e.g. `(color.array() * 255.0).min(255.0).max(0.0).round();`

the same applies to other code where we have `+0.5`

Reply:
ColorToUint8 and ColorToDouble conversions put in